python/hail/linalg/matrix.py

```diff
@@ -122,6 +162,42 @@ def __sub__(self, that):
     def dot(self, that):
         return BlockMatrix(self._jbm.multiply(that._jbm))
 
+    def entries_table(self):
+        """Returns a coordinate Table with a row for each entry in the BlockMatrix.
+
+        Examples
+        --------
+
+        >>> from hail.linalg import BlockMatrix
+        >>> import numpy as np
+        >>> block_matrix = BlockMatrix._from_numpy_matrix(np.matrix([[5,7],[2,8]]), 2)
+        >>> entries_table = block_matrix.entries_table()
+        >>> entries_table.show()
+        +--------+--------+-------------+
+        |      i |      j |       entry |
+        +--------+--------+-------------+
+        | !Int64 | !Int64 |    !Float64 |
+        +--------+--------+-------------+
+        |      0 |      0 | 5.00000e+00 |
+        |      0 |      1 | 7.00000e+00 |
+        |      1 |      0 | 2.00000e+00 |
+        |      1 |      1 | 8.00000e+00 |
+        +--------+--------+-------------+
+
+        Warning
+        -------
+        The table returned by this method should be used for aggregation or queries,
```

Comment:
I prefer:
"The resulting table may be filtered, aggregated, and queried, but should only be directly exported to disk if the block matrix is very small"